frontend/test/pytest/test_from_plxpr.py

```diff
@@ -0,0 +1,453 @@
+        assert converted.eqns[0].params["fn"] is circuit
+
+        catalyst_res = catalyst_execute_jaxpr(converted)(0.5)
+        assert len(catalyst_res) == 1
+        expected = np.array([np.cos(0.5 / 2) ** 2, np.sin(0.5 / 2) ** 2])
+        assert qml.math.allclose(catalyst_res[0], expected)
+
+        qjit_obj = qml.qjit(circuit)
+        qjit_obj(0.5)
+        catalxpr = qjit_obj.jaxpr
+        call_jaxpr_pl = converted.eqns[0].params["call_jaxpr"]
+        call_jaxpr_c = catalxpr.eqns[0].params["call_jaxpr"]
+
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c)
+
+    def test_state(self):
+        """Test that the state can be converted to catalxpr."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.Hadamard(0)
+            qml.IsingXX(phi, wires=(0, 1))
+            return qml.state()
+
+        phi = np.array(-0.6234)
+
+        qml.capture.enable()
+        plxpr = jax.make_jaxpr(circuit)(phi)
+        qml.capture.disable()
+
+        converted = from_plxpr(plxpr)(phi)
+
+        assert converted.eqns[0].primitive == catalyst.jax_primitives.func_p
+        assert converted.eqns[0].params["fn"] is circuit
+
+        catalyst_res = catalyst_execute_jaxpr(converted)(phi)
+        assert len(catalyst_res) == 1
+
+        x1 = np.cos(phi / 2) / np.sqrt(2)
+        x2 = -1j * np.sin(phi / 2) / np.sqrt(2)
+        expected = np.array([x1, x2, x1, x2])
+
+        assert qml.math.allclose(catalyst_res[0], expected)
+
+        qjit_obj = qml.qjit(circuit)
+        qjit_obj(phi)
+        catalxpr = qjit_obj.jaxpr
+        call_jaxpr_pl = converted.eqns[0].params["call_jaxpr"]
+        call_jaxpr_c = catalxpr.eqns[0].params["call_jaxpr"]
+
+        # confused by the weak_types error here
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c)
+
+    def test_variance(self):
+        """Test comparison and execution of a jaxpr containing a probability measurement."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.var(qml.Y(0))
+
+        x = np.array(0.724)
+
+        qml.capture.enable()
+        plxpr = jax.make_jaxpr(circuit)(x)
+        qml.capture.disable()
+
+        converted = from_plxpr(plxpr)(np.array(0.724))
+
+        assert converted.eqns[0].primitive == catalyst.jax_primitives.func_p
+        assert converted.eqns[0].params["fn"] is circuit
+
+        catalyst_res = catalyst_execute_jaxpr(converted)(x)
+        assert len(catalyst_res) == 1
+        expected = 1 - np.sin(x) ** 2
+        assert qml.math.allclose(catalyst_res[0], expected)
+
+        qjit_obj = qml.qjit(circuit)
+        qjit_obj(x)
+        catalxpr = qjit_obj.jaxpr
+        call_jaxpr_pl = converted.eqns[0].params["call_jaxpr"]
+        call_jaxpr_c = catalxpr.eqns[0].params["call_jaxpr"]
+
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c)
```

Comment:
Could parametrize the `expval, probs, var` tests, they are very similar.

Reply:
left as is, as I tend to favor verbose tests. Easy to debug later on.